service/security/test/src/test/resources/features/GroupServiceUnitTests.feature

```diff
@@ -0,0 +1,387 @@
+    And I search for the group with name "NewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNA"
+    Then I find the group with name "NewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNA"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group name to a too long one
+  Create an Access Group with valid name. Edit the group name to too long one, name that contains more than 255 characters.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalArgumentException" with the text "*"
+    When I update the group name to "NewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroupNewAccessGroup"
+    Then An exception was thrown
+
+  Scenario: Editing Access Group name to name with valid special symbols
+  Create an Access Group with valid name. Edit the group name to contain valid special symbols: "_", "-" and " ". Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "New Access_Group-ag"
+    And I search for the group with name "New Access_Group-ag"
+    Then I find the group with name "New Access_Group-ag"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group name to name with invalid special symbols in name
+  Create an Access Group with valid name. Edit the group name to name with invalid special symbols in name, all except "_", "-" and " ". Use one symbol every time.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalArgumentException" with the text "*"
+    When I update the group name to "NewAccessGroup@"
+    Then An exception was thrown
+
+  Scenario: Editing Access Group name to empty name
+  Create an Access Group with valid name. Edit the group name to empty name.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalNullArgumentException" with the text "*"
+    When I update the group name to " "
```

Comment:
Please change this step name to: "When I update the group name to """, becasue if you enter " " (with space) this is not empty, becasue it has one character (space). so it has te be "".